addons/website_mail_channel/models/mail_channel.py

```diff
@@ -27,3 +27,24 @@ def message_get_email_values(self, cr, uid, id, notif_mail=None, context=None):
         })
         res['headers'] = repr(headers)
         return res
+
+    @api.multi
+    def _send_confirmation_email(self, partner_ids, unsubscribe=False):
+        if unsubscribe:
+            template = self.env.ref('website_mail_channel.template_list_unsubscribe')
+            route = "/groups/unsubscribe/%s"
+        else:
+            template = self.env.ref('website_mail_channel.template_list_subscribe')
+            route = "/groups/subscribe/%s"
+
+        for partner in self.env['res.partner'].browse(partner_ids):
+            token = self.env['ir.token'].sudo().create({
+                'res_model': 'mail.channel',
+                'res_id': self.id,
+                'partner_id': partner.id,
+            })
+            template.with_context(
```

Comment:
There should be a value kwarg in template.send_mail to avoid that kind of context hack. However as it is currently not available, maybe do a send_mail with for_send = False, then update the mail recipients manually, instead of using the context to render it ? Not sure.

Anyway you can use recipient_ids instead of email_to = partner.email. It should display the name of the recipient like Recipient <email>, and is better.

An alternative is to set partner_to of the email template model to context['partner'].id .